src/Http/Http/src/StreamPipeWriter.cs

```diff
@@ -289,31 +303,44 @@ public void Dispose()
                 }
             }
 
-            _currentSegmentOwner?.Dispose();
+            DisposeOwner(_currentSegmentOwner);
+        }
+
+        private static void DisposeOwner(object owner)
```

Comment:
Helper method?

Reply:
To share with BufferSegment

Reply:
Same comment as above. Once we move this over to corefx then the merging will happen.